packages/effect/src/Chunk.ts

```diff
@@ -1387,3 +1387,53 @@ export const reduceRight: {
   <B, A>(b: B, f: (b: B, a: A, i: number) => B): (self: Chunk<A>) => B
   <A, B>(self: Chunk<A>, b: B, f: (b: B, a: A, i: number) => B): B
 } = RA.reduceRight
+
+/**
+ * Creates a `Chunk` of values not included in the other given `Chunk` using the provided `isEquivalent` function.
+ * The order and references of result values are determined by the first `Chunk`.
+ *
+ * @since 2.0.0
+ */
+export const differenceWith = <A>(isEquivalent: (self: A, that: A) => boolean): {
+  (that: Chunk<A>): (self: Chunk<A>) => Chunk<A>
+  (self: Chunk<A>, that: Chunk<A>): Chunk<A>
+} => {
+  return dual(
+    2,
+    (self: Chunk<A>, that: Chunk<A>): Chunk<A> => unsafeFromArray(RA.differenceWith(isEquivalent)(that, self))
+  )
+}
+
+/**
+ * Creates a `Chunk` of values not included in the other given `Chunk`.
+ * The order and references of result values are determined by the first `Chunk`.
+ *
+ * @since 3.2.0
+ */
+export const difference: {
+  <A>(that: Chunk<A>): (self: Chunk<A>) => Chunk<A>
+  <A>(self: Chunk<A>, that: Chunk<A>): Chunk<A>
+} = dual(
+  2,
+  <A>(self: Chunk<A>, that: Chunk<A>): Chunk<A> => unsafeFromArray(RA.difference(that, self))
+)
+
+/**
+ * Creates a `Chunk` of values not included in the other given `Chunk`.
+ * The order and references of result values are determined by the first `Chunk`.
+ *
+ * The first comparator `Chunk` may be wrapped in an `Option`. A `None` value is
+ * treated as an empty `Chunk` during comparison.
+ *
+ * This is primarily for convenience on zip operations, gets, or similar.
+ *
+ * @since 3.2.0
+ */
+export const differenceOption: {
+  <A>(that: Chunk<A>): (self: Option<Chunk<A>>) => Chunk<A>
+  <A>(self: Option<Chunk<A>>, that: Chunk<A>): Chunk<A>
+} = dual(
+  2,
+  <A>(self: Option<Chunk<A>>, that: Chunk<A>): Chunk<A> =>
+    unsafeFromArray(RA.difference(that, O.getOrElse(self, empty)))
+)
```

Comment:
I'm always on the fence when it comes to adding additional permutations / combinations like these that can be easily composed in userland. It always raises the question "where do we stop?". E.g. here we could also add a variant for `Either` and `Exit`, etc.

Even though that is also true for the other two, I can see the value in those, but I would leave this one out.